docs/adr/000x-modelling-history.md

```diff
@@ -0,0 +1,250 @@
+to directly map the concept of a revision to each time a user consciously
+revises a document. The [anatomy of a Revision model](#breakdown-of-revision)
+is explored further in this document.
+
+**Status**: Represents a state that an edition can hold such as: "draft" or
+"submitted for review". This model is coupled to the concept of status that is
+shown and changed by a user. Each time a user changes the status of an edition
+a new Status model is created and the user who created it stored. An edition
+can only have one status at any one time. It is intended to associate with
+specialist data for a state such as a Removal or Withdrawal, which allows the
+history of these to no longer be the responsibility of TimelineEntry. Initially
+this object is intended to be immutable however this may be changed if status
+changes become asynchronous operations, which may be the case with scheduling.
+
+### Approach to mutability/immutability
+
+A number of the models in Content Publisher are defined as immutable, most
+significantly Revision and its associated types. These models should be
+persisted to the database once and never updated or deleted, any need to change
+them requires creating a new record. This allows us to store a full history
+by only appending to the database.
+
+Accessing immutable models is intended to always be done by a foreign key and
+not by the usage of `SELECT MAX` style queries. This is done for
+simplicity, performance and for consistency with Rails idioms. To achieve this
+there are a some mutable models, eg Edition, that reference immutable models,
+eg Revision. When the content for an edition is revised the Edition model
+references the new Revision. Mutable models should not be used to store fields
+where there is a need to know how the value of that field changed over time.
+
+The choice of this immutability strategy is to store both present and
+historical concerns in the same data stores, thus ensuring history remains a
+first class citizen. A nice side effect of having immutable models is
+this opens options for caching. Since data for that
+model will never change it can effectively be cached forever.
+
+### Breakdown of Revision
+
+Revision is a model that stores little data itself and joins to other models
+that store the data. This can be visualised as:
+
+![Revision breakdown](000x/revision-diagram.png)
+
+The intention of breaking this up is to be conservative with the amount of data
+duplicated for both storage and maintenance reasons. Thus when a user edits
+the title of an edition a new ContentRevision object is created and the existing
+objects (TagsRevision, MetadataRevision and ImageRevisions) are associated with
+the next revision. An ImageRevision is modelled in a similar way to a Revision
+and this is explained further in [Image modelling](#image-modelling).
+
+It is intended that [delegation][delegate] be used when interfacing with a
```

Comment:
Should we also mention that we intend to delegate update operations, for the same reason?

Reply:
Oh sorry - I thought this was in reference to the models section in consequences which I updated.

Looking at this I don't think it needs changing as it only references interfacing which can refer to both reading or writing.